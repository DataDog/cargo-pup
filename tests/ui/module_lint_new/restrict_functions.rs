//@compile-flags: --crate-name test_restrict_functions
//@compile-flags: --crate-type lib

// This test verifies that the ModuleRule::RestrictFunctions works correctly
// by flagging functions that exceed the maximum allowed length (3 lines in this case)

// The following function is too long (6 lines including signature and closing brace)
// It should trigger the lint with the maximum length of 3 lines
fn too_long_function() { //~ ERROR: Function exceeds maximum length of 3 lines with 6 lines
    let a = 1;
    let b = 2;
    let c = 3;
    println!("This function has too many lines");
}

// This function is ok (3 lines including signature and closing brace)
fn short_function() {
    println!("This function is OK");
}

// Ensure the lint also works for methods inside impls
pub struct TestStruct;

impl TestStruct {
    // This method is too long (6 lines)
    fn too_long_method(&self) { //~ ERROR: Function exceeds maximum length of 3 lines with 6 lines
        let x = 1;
        let y = 2;
        let z = 3;
        println!("This method has too many lines");
    }
    
    // This method is OK (3 lines)
    fn short_method(&self) {
        println!("This method is OK");
    }
} 