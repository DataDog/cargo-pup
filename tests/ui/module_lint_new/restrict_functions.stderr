error: Function exceeds maximum length of 3 lines with 6 lines
  --> tests/ui/module_lint_new/restrict_functions.rs:9:1
   |
LL | / fn too_long_function() {
LL | |     let a = 1;
LL | |     let b = 2;
LL | |     let c = 3;
LL | |     println!("This function has too many lines");
LL | | }
   | |_^
   |
   = help: Consider breaking this function into smaller parts
   = note: Applied by cargo-pup rule 'restrict_functions_test'.
   = note: `#[deny(module_lint)]` on by default

error: Function exceeds maximum length of 3 lines with 6 lines
  --> tests/ui/module_lint_new/restrict_functions.rs:26:5
   |
LL | /     fn too_long_method(&self) {
LL | |         let x = 1;
LL | |         let y = 2;
LL | |         let z = 3;
LL | |         println!("This method has too many lines");
LL | |     }
   | |_____^
   |
   = help: Consider breaking this method into smaller parts
   = note: Applied by cargo-pup rule 'restrict_functions_test'.

error: aborting due to 2 previous errors

